sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md

```diff
@@ -306,7 +317,58 @@ await foreach (PartitionEvent receivedEvent in consumer.ReadEventsFromPartitionA
 // This is where applications can verify that the partition events output by the consumer client were handled as expected.
 ```
 
-## Reading properties from `EventHubProducerClient`, `EventHubBufferedProducerClient`, and `EventHubConsumerClient`
+## Consuming events using the `PartitionReceiver`
+
+The sample below illustrates how to mock a `PartitionReceiver`, and set up its `ReceiveBatchAsync` method to return a simple data batch. 
+
+Given the purpose of the `PartitionReceiver`, it is anticipated that most applications will have complex code surrounding their receivers, so this snippet is intentionally simple, with the focus being on using the `EventHubsModelFactory` to mock events being returned from the broker.
+
+```C# Snippet:EventHubs_Sample11_PartitionReceiverMock
+Mock<PartitionReceiver> mockReceiver = new();
+
+// ReceivedEvents is a list of events to use when mocking ReceiveBatchAsync within the PartitionReceiver.
+
+List<EventData> receivedEvents = new();
+
+for (int index = 0; index < 10; index++)
+{
+    // Mocking an EventData instance, different arguments can simulate different potential
+    // outputs from the broker
+    EventData eventData = EventHubsModelFactory.EventData(
+        eventBody: new BinaryData($"Sample-Event-{index}"),
+        systemProperties: new Dictionary<string, object>(), //arbitrary value
+        partitionKey: $"sample-key-{index}",
+        sequenceNumber: 1234,
+        offset: 234,
+        enqueuedTime: DateTimeOffset.Parse("9:25 AM"));
+
+    receivedEvents.Add(eventData);
+}
+
+// This sets up the mock to receive the list of events defined above when ReceiveBatchAsync is called
+// on a given partition.
+
+mockReceiver
+    .Setup(r => r.ReceiveBatchAsync(
+        It.IsAny<int>(),
+        It.IsAny<TimeSpan>(),
+        It.IsAny<CancellationToken>()))
+    .ReturnsAsync(receivedEvents);
+
+// The following demonstrates how to use the mocked methods. This would be where application code utilizing
+// the mocked methods could be called and verified.
+
+PartitionReceiver receiver = mockReceiver.Object;
+
+using CancellationTokenSource cancellationTokenSource = new();
+
+IEnumerable<EventData> receivedBatch = await receiver.ReceiveBatchAsync(
+    maximumEventCount: 10,
+    TimeSpan.FromSeconds(1),
+    cancellationTokenSource.Token);
+```
+
+## Querying metadata properties using the `EventHubProducerClient`, `EventHubBufferedProducerClient`, or `EventHubConsumerClient`
```

Comment:
I like this title more too!

Reply:
I was trolling through other samples and realized this is how we seem to refer to it elsewhere.